File: 009 - Palindrome Number.cpp
// Given an integer NUMBER, return true if its palidrome and false otherwise

/*
Solution: two pointers START and END. START starting from the element and incrementing, END starting from the last element and decremeneting. 
If NUMBER[START] != NUMBER[END] return false, otherwise its true

Complexity:
- Time: O(n)
*/

var isPalindrome = function(x) {
    const NUMBER = String(x);
    let start = 0;
    let end = NUMBER.length-1;
    while (start < end){
        if(NUMBER[start]!== NUMBER[end] )return false
        start++;
        end--;
    }
    return true

};

File: 009 - Palindrome Number.go
func isPalindrome(x int) bool {
    if x < 0 {
        return false
    }

    if x < 10{
        return true
    }

    if x%10 == 0 {
        return false
    }

    aux := 0
    for x > 0 {
        aux = aux*10+x%10
        x /= 10

        if aux == x || aux == x/10 {
            return true
        }
    }

    return false
}
